s3bucket/main.tf

```diff
@@ -0,0 +1,57 @@
+resource "aws_s3_bucket" "codedeploy_bucket" {
+  bucket = "${var.name_prefix}-codedeploy-releases"
+  acl    = "private"
+
+  versioning {
+    enabled = true
+  }
+}
+
+resource "aws_iam_policy" "codedeploy_policy" {
```

Comment:
That's a different thing, that is to create the role that the codedeploy service will use. This is the IAM policy that's attached to the EC2 instances deployed with codedeploy, so it can access the S3 bucket where the release packages are.